packages/nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -242,6 +242,255 @@ contract MultiAssetHolder is IMultiAssetHolder, StatusManager {
         );
     }
 
+    /**
+     * @notice Transfers as many funds escrowed against `sourceChannelId` as can be afforded for the destinations specified by indices in the beneficiaries of the __target__ of the channel at indexOfTargetInSource.
+     * @dev Transfers as many funds escrowed against `sourceChannelId` as can be afforded for the destinations specified by indices in the beneficiaries of the __target__ of the channel at indexOfTargetInSource.
+     * @param assetIndex the index of the targetted asset
+     * @param claimArgs arguments used in the claim function. Used to avoid stack too deep error.
+     */
+    function claim(uint256 assetIndex, ClaimArgs memory claimArgs) external override {
+        Outcome.SingleAssetExit[] memory sourceOutcome;
+        Outcome.SingleAssetExit[] memory targetOutcome;
+        uint256 initialAssetHoldings;
+        {
+            bytes memory sourceOutcomeBytes = claimArgs.sourceOutcomeBytes;
+            bytes memory targetOutcomeBytes = claimArgs.targetOutcomeBytes;
+            (sourceOutcome, targetOutcome, initialAssetHoldings) = _apply_claim_checks(
+                assetIndex,
```

Comment:
Would it be cleaner to just pass in `ClaimArgs`? But perhaps that uses more gas.

Reply:
It would be cleaner, I think -- I went ahead and did that 42dd3b6.